pkg/cmd/config/list/list.go

```diff
@@ -0,0 +1,89 @@
+package list
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc/v2"
+	"github.com/OctopusDeploy/cli/pkg/constants"
+	"github.com/OctopusDeploy/cli/pkg/factory"
+	"github.com/OctopusDeploy/cli/pkg/output"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func NewCmdList(_ factory.Factory) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List values from config file",
+		Long:  "List values from config file.",
+		Example: fmt.Sprintf(heredoc.Doc(`
+			$ %s config list"
+		`), constants.ExecutableName),
+		Aliases: []string{"ls"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return listRun(cmd)
+		},
+	}
+
+	return cmd
+}
+
+func listRun(cmd *cobra.Command) error {
+	configFile := viper.New()
+	configFile.SetConfigFile(viper.ConfigFileUsed())
+	configFile.ReadInConfig()
+
+	type ConfigData struct {
+		ApiKey       string `json:"apikey"`
+		Editor       string `json:"editor"`
+		Host         string `json:"host"`
+		NoPrompt     string `json:"noprompt"`
+		OutputFormat string `json:"outputformat"`
+		Space        string `json:"space"`
+	}
+
+	outputFormat, _ := cmd.Flags().GetString(constants.FlagOutputFormat)
+	if outputFormat == "" {
+		outputFormat = viper.GetString(constants.ConfigOutputFormat)
+	}
+
+	switch strings.ToLower(outputFormat) {
+	case constants.OutputFormatJson:
+		configData := &ConfigData{}
+		for _, key := range configFile.AllKeys() {
+			switch strings.ToLower(key) {
+			case strings.ToLower(constants.ConfigApiKey):
+				configData.ApiKey = configFile.GetString(key)
+			case strings.ToLower(constants.ConfigEditor):
+				configData.Editor = configFile.GetString(key)
+			case strings.ToLower(constants.ConfigHost):
+				configData.Host = configFile.GetString(key)
+			case strings.ToLower(constants.ConfigNoPrompt):
+				configData.NoPrompt = configFile.GetString(key)
+			case strings.ToLower(constants.ConfigSpace):
+				configData.Space = configFile.GetString(key)
+			case strings.ToLower(constants.ConfigOutputFormat):
+				configData.OutputFormat = configFile.GetString(key)
+			default:
+				return fmt.Errorf("the key '%s' is not a supported config option", key)
+			}
+		}
+		data, _ := json.MarshalIndent(configData, "", "  ")
+		cmd.Println(string(data))
+	case constants.OutputFormatBasic:
+		for _, key := range configFile.AllKeys() {
+			cmd.Println(configFile.GetString(key))
+		}
+	case constants.OutputFormatTable:
+		t := output.NewTable(cmd.OutOrStdout())
+		t.AddRow(output.Bold("KEY"), output.Bold("VALUE"))
+		for _, key := range configFile.AllKeys() {
+			t.AddRow(key, configFile.GetString(key))
```

Comment:
This is going to print the API key directly to the screen.
I feel like we should print ****s instead for this (same deal with JSON), but I'm not sure if I'm over-thinking it. 

Right now, the API key is trivially readable from the JSON file, so printing it to the screen isn't much worse -- but we have discussed things like putting the API key in the windows/macos keystore, and printing those values isn't something you'd want to be doing normally.

Would be good to get @jbristowe 's opinion on this one.

Reply:
I'm inclined to agree here. Maybe we should do what they do with credit card numbers; show the last 4 digits and print `*` for the rest.

Reply:
As already said. It's still plain text in the json and the `config get apikey` command will still return the text value. I hid it anyway in the list command, as I could see "accidental" exposing the  key here as an issue as where doing it as `config get apikey` is it explicit.